### PR TITLE
Generalised wrapper for sklearn classifiers

### PR DESCRIPTION
**Pull request by tblah**
Sentences are padded to a configurable, fixed length.

If unknown words are found by Word2Vec when encoding a token, an unknown symbol is inserted. 

Adding and testing new sklearn classifiers is just a couple of lines of code. Testing is just raw accuracy at the moment because the dataset seems quite evenly split between each label and RandomForest (most of the testing) isn't working well enough to matter anyway.

Examples of a random forest and a linear (no kernel but faster) svm are included. 

The stuff under data/ would probably be useful for tensorflow stuff too.

**Edit 11/04:** See the branch for newer commits which report better statistics and does randomised parameter search

### CODE REVIEW

**Comment by superdyoll** [comment]
Do you still want this pull request to be code reviewed given the edit? I think the request has all of your latest commits

**Comment by tblah** [comment]
@superdyoll yeah it seems to be tracking new commits as I put them into the branch. This is what I want. Please merge whatever the current state of the branch is when you get around to it. I reviewed the automated code review comments on 6b7eba7 and imho they should not be changed.

**Comment by tblah** [comment]
To be clear, at minimum, the assertions should definitely not be changed according to the code review. I was intending for them not to appear in optimised code. I suppose they could be removed if people would prefer that. The import statement is for alternative examples which I didn't happen to be testing at the time of the commit. It would be much more convenient if it remained so that the other label encoding can be tested by only uncommenting a single line in one place instead of having to remember to also uncomment the import statement. We don't try everything on every run because that would take hours.

File: Predictors/sklearn.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+from sklearn.model_selection import cross_val_predict, RandomizedSearchCV
+from sklearn.metrics import confusion_matrix, precision_score, recall_score, f1_score, accuracy_score
+
+import numpy as np
+
+# doens't really do much. See data.{padded_sentences, named_authors} for
+# data processing
+
+
+class SklearnClassifier(ABC, BaseEstimator, ClassifierMixin):
```
Comment (superdyoll) [inline Predictors/sklearn.py]: As the file is called sklearn causes the module sklearn not to picked up by other classes. Suggest renaming file to sklearnclassifier or something like that.

**Review by superdyoll — CHANGES_REQUESTED** [review]
Once the file name is changed looks good to go

**Review by superdyoll — APPROVED** [review]
